### PR TITLE
Implement support for markup in wxDataViewCtrl

### PR DESCRIPTION
**Pull request by vadz**
Add `wxDataViewTextRenderer::EnableMarkup()` for finer-grained control over items appearance.

### CODE REVIEW

**Comment by discnl** [comment]
I only skimmed through the changes and have some minor typo/style remarks:

In the commit message of 0c7424f22ebd0385d7dc98b720aabcf8343e4e59 `evne` should be `even`.

In the changes of the other two commits I think "\* " should be replaced by " \*", e.g. `wxWindow* win` becomes `wxWindow *win`. Unless it's OK to introduce that style in new wx code?

File: interface/wx/dataview.h [hunk 1]
```diff
@@ -1966,6 +1966,49 @@ class wxDataViewTextRenderer : public wxDataViewRenderer
     wxDataViewTextRenderer(const wxString& varianttype = GetDefaultType(),
                            wxDataViewCellMode mode = wxDATAVIEW_CELL_INERT,
                            int align = wxDVR_DEFAULT_ALIGNMENT );
+
+    /**
+        Enable interpretation of markup in the cells data.
```
Comment (discnl) [inline interface/wx/dataview.h]: `cells'` or `cell's` ?

**Comment by vadz** [comment]
Personally I use `wxWindow*` because it's consistent with e.g. `wxDC&` and also because it's the only form that can really be used as `wxWindow* const` (`wxWindow *const` is plain weird IMO), but yes, you're probably right and we shouldn't use it in new wx code, I've fixed all the occurrences of it except for the one with `const`.

Also fixed the typo and changed the docs to say just "data", "cell" is inconsistent with "data of the items" used below and the latter is arguably more clear, even more verbose.

**Comment by discnl** [comment]
> Personally I use wxWindow* because it's consistent with e.g. wxDC& and also because it's the only form that can really be used as wxWindow* const (wxWindow *const is plain weird IMO), but yes, you're probably right and we shouldn't use it in new wx code, I've fixed all the occurrences of it except for the one with const.

I'm fine with any style: no strict rules for these minor styles, or sticking to what is mostly used in wx, or having some newer preference for newly added code. It's more about knowing which one to follow. For example for the pointers the guidelines don't mention anything, at best some unrelated function declaration mentioning `const char *text`.

File: src/generic/markuptext.cpp [hunk 2]
```diff
@@ -217,12 +190,96 @@ class wxMarkupParserRenderOutput : public wxMarkupParserAttrOutput
...
+                                 wxALIGN_LEFT | wxALIGN_CENTRE_VERTICAL,
+                                 m_rendererFlags,
+                                 wxELLIPSIZE_NONE);
+
+        m_pos += m_dc.GetTextExtent(text).x;
+    }
+
+private:
+    wxWindow* const m_win;
+    int const m_rendererFlags;
```
Comment (discnl) [inline src/generic/markuptext.cpp]: `const int`? At least it's a way more common style in wx.

Comment (vadz) [inline src/generic/markuptext.cpp]: Yes, but it's inconsistent with the lines just above and below it...

Comment (discnl) [inline src/generic/markuptext.cpp]: I noticed that but forgot to mention it as a reason. Alrighty then :-).

**Comment by vadz** [comment]
De facto, the existing code uses `T *` much more than `T*`, but the latter makes more sense (because of `T&` and `T* const` considerations), so I'm just not sure.

**Comment by vadz** [comment]
Merged in f792ba45f6e062fbea699aeb43e126df1e78b605

**Comment by iwbnwif** [comment]
This is a great enhancement, thanks!

Might it be worth populating the model in the sample differently if wxUSE_MARKUP is not set? This is because currently the sample shows the entire markup in the "attributes" column which is a bit untidy.

**Comment by vadz** [comment]
I thought it wasn't worth it because `wxUSE_MARKUP` was going to be 1 in 99% of cases anyhow. If you explicitly disable it, I don't think it's such a big deal to see what you're missing in the sample :-)

**Pull request closed without merging**